src/main/java/org/truffleruby/core/exception/SyntaxErrorNodes.java

```diff
@@ -35,4 +40,26 @@ RubySyntaxError allocateSyntaxError(RubyClass rubyClass) {
 
     }
 
+    @CoreMethod(names = "path")
+    public abstract static class PathNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        RubyString path(RubySyntaxError syntaxError) {
+            if (!syntaxError.hasSourceLocation()) {
+                return coreStrings().UNKNOWN.createInstance(getContext());
+            }
+
+            Source source;
+            try {
+                source = syntaxError.getSourceLocation().getSource();
+            } catch (UnsupportedMessageException e) {
+                throw CompilerDirectives.shouldNotReachHere(e);
+            }
+
+            var path = getLanguage().getPathToTStringCache().getCachedPath(source);
+            return createString(path, Encodings.UTF_8);
```

Comment:
This code is mostly copied from https://github.com/shopify/truffleruby/blob/add-syntaxerror-path/src/main/java/org/truffleruby/interop/SourceLocationNodes.java#L42-L56 (thanks @nirvdrum for the pointers!), but I'm not sure what's the ideal way to DRY this out (naively, could we move this into `SourceSection#getSource?`)

Reply:
There is very little logic reused here so it seems fine as-is